kani-driver/src/cbmc_output_parser.rs

```diff
@@ -0,0 +1,989 @@
+        result: Vec<Property>,
+    },
+    #[serde(rename_all = "camelCase")]
+    ProverStatus {
+        _c_prover_status: String,
+    },
+}
+
+impl ParserItem {
+    /// Determines if an item must be skipped or not.
+    fn must_be_skipped(&self) -> bool {
+        matches!(&self, ParserItem::Message { message_text, .. } if message_text.starts_with("Building error trace") || message_text.starts_with("VERIFICATION"))
+    }
+}
+
+/// Struct that represents a property.
+///
+/// Note: `reach` is not part of the parsed data, but it's useful to annotate
+/// its reachability status.
+#[derive(Clone, Debug, Deserialize)]
+pub struct Property {
+    pub description: String,
+    pub property: String,
+    #[serde(rename = "sourceLocation")]
+    pub source_location: SourceLocation,
+    pub status: CheckStatus,
+    pub reach: Option<CheckStatus>,
+    pub trace: Option<Vec<TraceItem>>,
+}
+
+/// Struct that represents a source location.
+///
+/// Source locations may be completely empty, which is why
+/// all members are optional.
+#[derive(Clone, Debug, Deserialize)]
+pub struct SourceLocation {
+    pub column: Option<String>,
+    pub file: Option<String>,
+    pub function: Option<String>,
+    pub line: Option<String>,
+}
+
+impl SourceLocation {
+    /// Determines if fundamental parts of a source location are missing.
+    fn is_missing(&self) -> bool {
+        self.file.is_none() && self.function.is_none()
+    }
+}
+
+/// `Display` implement for `SourceLocation`.
+///
+/// This is used to format source locations for individual checks. But source
+/// locations may be printed in a different way in other places (e.g., in the
+/// "Failed Checks" summary at the end).
+///
+/// Source locations formatted this way will look like:
+/// `<file>:<line>:<column> in function <function>`
+/// if all attributes were specified. Otherwise, we:
+///  * Omit `in function <function>` if the function isn't specified.
+///  * Use `Unknown file` instead of `<file>:<line>:<column>` if the file isn't
+///    specified.
+///  * Lines and columns are only formatted if they were specified and preceding
+///    attribute was formatted.
+impl std::fmt::Display for SourceLocation {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        use std::fmt::Write;
+        let mut fmt_str = String::new();
+        if let Some(file) = self.file.clone() {
+            let file_path = filepath(file);
+            write!(&mut fmt_str, "{file_path}")?;
+            if let Some(line) = self.line.clone() {
+                write!(&mut fmt_str, ":{line}")?;
+                if let Some(column) = self.column.clone() {
+                    write!(&mut fmt_str, ":{column}")?;
+                }
+            }
+        } else {
+            write!(&mut fmt_str, "Unknown file")?;
+        }
+        if let Some(function) = self.function.clone() {
+            write!(&mut fmt_str, " in function {function}")?;
+        }
+
+        write! {f, "{}", fmt_str}
```

Comment:
Since we're first writing to `fmt_str` and then write `fmt_str` to `f`, can't we just write to `f` directly?